app/services/post_check_availability_task_service.rb

```diff
@@ -0,0 +1,48 @@
+class PostCheckAvailabilityTaskService < TaskService
+  def initialize(options)
+    super
+    @source = Source.find(options[:source_id]) || raise("source_id must be specified in options")
+  end
+
+  def process
+    update_source
+    create_kafka_event
+    create_refresh_upload_task
```

Comment:
@hsong-rh This should be conditional only if the source is available, if the source is not available we should skip this code with a warning saying upload has not been run